src/utils/misc.ts

```diff
@@ -0,0 +1,7 @@
+import { AssertionError } from 'assert'
+
+export function assert<T>(val: T, message: string): asserts val is NonNullable<T> {
+  if (!val) {
```

Comment:
This isn't exactly a NonNullable assertion, though it may be ok for our purposes
`!val` will break on every falsy value (`0, ''`)

https://www.typescriptlang.org/docs/handbook/utility-types.html#nonnullablet

Reply:
Any suggestion on how to fix this?

Reply:
you can do sth like `val ?? somethingElse` where `0` and `''` will be accepted but nothing else

nullish coalescense or sth similar in TS 3.7+